### PR TITLE
nginx: reduced CLI logging when reloading nginx

### LINKED ISSUE #8231: Stop printing nginx: [error] invalid PID number "" in "/run/nginx.pid"
If you use Certbot to obtain a certificate using the nginx plugin and nginx is initially off, you should see output like:
```
nginx: [error] invalid PID number "" in "/run/nginx.pid"
```
As part of our results from user testing, we should remove this output.

This should be verified, but my understanding here is this comes from [this call to `nginx -s reload`](https://github.com/certbot/certbot/blob/2a047eb526ee02d4bb3bb3668260ac6d481b99d3/certbot-nginx/certbot_nginx/_internal/configurator.py#L1178-L1179) where stdout/stderr isn't captured. This command fails and then we try starting nginx which works.

If this understanding is correct, I think we should capture the output from the first command and either:

1. Log it at the debug level.
2. Hold it in a local variable to be logged/raised in an exception if the command starting nginx also fails.

### PR DESCRIPTION
**Pull request by alexzorin**
Hides the output of `nginx -s reload` from the CLI, moving it to
debug-level logging.

Additionally, fixes an issue where Certbot did not properly capture the
output of the nginx reload and restart commands.

Fixes #8231

### CODE REVIEW

File: certbot-nginx/tests/configurator_test.py [hunk 1]
```diff
@@ -466,14 +466,18 @@ def test_nginx_restart(self, mock_time, mock_popen):
         mocked.communicate.return_value = ('', '')
         mocked.returncode = 0
         self.config.restart()
+        assert mocked.communicate.call_count == 1
```
Comment (ohemorange) [inline certbot-nginx/tests/configurator_test.py:469]: let's use `self.assertEqual(a, b)` here

File: certbot-nginx/tests/configurator_test.py [hunk 2]
```diff
@@ -466,14 +466,18 @@ def test_nginx_restart(self, mock_time, mock_popen):
...
 
     @mock.patch("certbot_nginx._internal.configurator.subprocess.Popen")
-    def test_nginx_restart_fail(self, mock_popen):
+    @mock.patch("certbot_nginx._internal.configurator.logger.debug")
+    def test_nginx_restart_fail(self, mock_log_debug, mock_popen):
         mocked = mock_popen()
         mocked.communicate.return_value = ('', '')
         mocked.returncode = 1
         self.assertRaises(errors.MisconfigurationError, self.config.restart)
+        assert mocked.communicate.call_count == 2
```
Comment (ohemorange) [inline certbot-nginx/tests/configurator_test.py:479]: let's use `self.assertEqual(a, b)` here

**Review by ohemorange — CHANGES_REQUESTED** [review]
Do you have a before and after of certbot's output?

**Comment by alexzorin** [comment]
> Do you have a before and after of certbot's output?

**Scenario 1**: When nginx is stopped, but can be started:

Effect: the output from nginx is muted entirely, because the fallback restart strategy succeeded.

Before:

    $ sudo certbot install --nginx --cert-name random27579.example.org
    Saving debug log to /var/log/letsencrypt/letsencrypt.log
    Plugins selected: Authenticator None, Installer nginx
    Deploying Certificate to VirtualHost /etc/nginx/sites-enabled/default
    Deploying Certificate to VirtualHost /etc/nginx/sites-enabled/default
    nginx: [error] invalid PID number "" in "/run/nginx.pid"
    Redirecting all traffic on port 80 to ssl in /etc/nginx/sites-enabled/default
    Redirecting all traffic on port 80 to ssl in /etc/nginx/sites-enabled/default

After:

    $ sudo certbot install --nginx --cert-name random27579.example.org
    Saving debug log to /var/log/letsencrypt/letsencrypt.log
    Plugins selected: Authenticator None, Installer nginx
    Deploying Certificate to VirtualHost /etc/nginx/sites-enabled/default
    Deploying Certificate to VirtualHost /etc/nginx/sites-enabled/default
    Redirecting all traffic on port 80 to ssl in /etc/nginx/sites-enabled/default
    Redirecting all traffic on port 80 to ssl in /etc/nginx/sites-enabled/default

---

**Scenario 2**: When nginx is stopped and cannot be started:

Effect: the error output is suppressed, captured and emitted later.

Before:

    $ sudo certbot install --nginx --cert-name random27579.example.org
    Saving debug log to /var/log/letsencrypt/letsencrypt.log
    Plugins selected: Authenticator None, Installer nginx
    Deploying Certificate to VirtualHost /etc/nginx/sites-enabled/default
    Deploying Certificate to VirtualHost /etc/nginx/sites-enabled/default
    nginx: [alert] kill(99108, 1) failed (3: No such process)
    Rolling back to previous server configuration...
    nginx: [alert] kill(99108, 1) failed (3: No such process)
    Encountered exception during recovery:
    Traceback (most recent call last):
      File "/home/alex/devel/certbot/certbot/certbot/_internal/client.py", line 530, in deploy_certificate
        self.installer.restart()
      File "/home/alex/devel/certbot/certbot-nginx/certbot_nginx/_internal/configurator.py", line 918, in restart
        nginx_restart(self.conf('ctl'), self.nginx_conf, self.conf('sleep-seconds'))
      File "/home/alex/devel/certbot/certbot-nginx/certbot_nginx/_internal/configurator.py", line 1193, in nginx_restart
        raise errors.MisconfigurationError(
    certbot.errors.MisconfigurationError: nginx restart failed:
    b''
    b''

    During handling of the above exception, another exception occurred:

    Traceback (most recent call last):
      File "/home/alex/devel/certbot/certbot/certbot/_internal/error_handler.py", line 125, in _call_registered
        self.funcs[-1]()
      File "/home/alex/devel/certbot/certbot/certbot/_internal/client.py", line 630, in _rollback_and_restart
        self.installer.restart()
      File "/home/alex/devel/certbot/certbot-nginx/certbot_nginx/_internal/configurator.py", line 918, in restart
        nginx_restart(self.conf('ctl'), self.nginx_conf, self.conf('sleep-seconds'))
      File "/home/alex/devel/certbot/certbot-nginx/certbot_nginx/_internal/configurator.py", line 1193, in nginx_restart
        raise errors.MisconfigurationError(
    certbot.errors.MisconfigurationError: nginx restart failed:
    b''
    b''
    nginx restart failed:
    b''
    b''

    IMPORTANT NOTES:
    - An error occurred and we failed to restore your config and restart
      your server. Please post to
      https://community.letsencrypt.org/c/help with details about your
      configuration and this error you received.

After:

    $ sudo certbot install --nginx --cert-name random27579.example.org
    Saving debug log to /var/log/letsencrypt/letsencrypt.log
    Plugins selected: Authenticator None, Installer nginx
    Deploying Certificate to VirtualHost /etc/nginx/sites-enabled/default
    Deploying Certificate to VirtualHost /etc/nginx/sites-enabled/default
    Rolling back to previous server configuration...
    Encountered exception during recovery:
    Traceback (most recent call last):
      File "/home/alex/devel/certbot/certbot/certbot/_internal/client.py", line 530, in deploy_certificate
        self.installer.restart()
      File "/home/alex/devel/certbot/certbot-nginx/certbot_nginx/_internal/configurator.py", line 918, in restart
        nginx_restart(self.conf('ctl'), self.nginx_conf, self.conf('sleep-seconds'))
      File "/home/alex/devel/certbot/certbot-nginx/certbot_nginx/_internal/configurator.py", line 1199, in nginx_restart
        raise errors.MisconfigurationError(
    certbot.errors.MisconfigurationError: nginx restart failed:
    nginx: [emerg] bind() to 0.0.0.0:80 failed (98: Address already in use)
    nginx: [emerg] bind() to 0.0.0.0:80 failed (98: Address already in use)
    nginx: [emerg] bind() to 0.0.0.0:80 failed (98: Address already in use)
    nginx: [emerg] bind() to 0.0.0.0:80 failed (98: Address already in use)
    nginx: [emerg] bind() to 0.0.0.0:80 failed (98: Address already in use)
    nginx: [emerg] still could not bind()


    During handling of the above exception, another exception occurred:

    Traceback (most recent call last):
      File "/home/alex/devel/certbot/certbot/certbot/_internal/error_handler.py", line 125, in _call_registered
        self.funcs[-1]()
      File "/home/alex/devel/certbot/certbot/certbot/_internal/client.py", line 630, in _rollback_and_restart
        self.installer.restart()
      File "/home/alex/devel/certbot/certbot-nginx/certbot_nginx/_internal/configurator.py", line 918, in restart
        nginx_restart(self.conf('ctl'), self.nginx_conf, self.conf('sleep-seconds'))
      File "/home/alex/devel/certbot/certbot-nginx/certbot_nginx/_internal/configurator.py", line 1199, in nginx_restart
        raise errors.MisconfigurationError(
    certbot.errors.MisconfigurationError: nginx restart failed:
    nginx: [emerg] bind() to 0.0.0.0:80 failed (98: Address already in use)
    nginx: [emerg] bind() to 0.0.0.0:80 failed (98: Address already in use)
    nginx: [emerg] bind() to 0.0.0.0:80 failed (98: Address already in use)
    nginx: [emerg] bind() to 0.0.0.0:80 failed (98: Address already in use)
    nginx: [emerg] bind() to 0.0.0.0:80 failed (98: Address already in use)
    nginx: [emerg] still could not bind()

    nginx restart failed:
    nginx: [emerg] bind() to 0.0.0.0:80 failed (98: Address already in use)
    nginx: [emerg] bind() to 0.0.0.0:80 failed (98: Address already in use)
    nginx: [emerg] bind() to 0.0.0.0:80 failed (98: Address already in use)
    nginx: [emerg] bind() to 0.0.0.0:80 failed (98: Address already in use)
    nginx: [emerg] bind() to 0.0.0.0:80 failed (98: Address already in use)
    nginx: [emerg] still could not bind()


    IMPORTANT NOTES:
    - An error occurred and we failed to restore your config and restart
      your server. Please post to
      https://community.letsencrypt.org/c/help with details about your
      configuration and this error you received.


---

We should also inhibit the "During handling of the above exception, another exception occurred" since it is entirely duplicative, but probably as part of a different change.

**Review by ohemorange — APPROVED** [review]
> We should also inhibit the "During handling of the above exception, another exception occurred" since it is entirely duplicative, but probably as part of a different change.

Agreed; could you make an issue for that?

Otherwise, lgtm!

**Pull request merged**